### PR TITLE
Flag / Link / Image-related Academy Updates

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-1955?filter=-1

## Summary
Small set of related updates to Link / Flag / Image based on several roadblocks I ran into while building out the new Academy homepage (specifically the [Flags](https://epic-academy.boltdesignsystem.com/pattern-lab/?p=pages-homepage) used on the homepage)

## Details
Link-specific updates
- Added new `block` display option to Link
- Added max-width CSS fix to the link's inner text element 

Image-specific updates
- Image component offsetWidth JS bug fix I ran into while these updates were being made

Flag-specific updates
  - Switched Flag twig template to use Display Block links (instead of display Flex links) 

## How to test
- Review changes

### CODE REVIEW

File: packages/components/bolt-link/src/link.js [hunk 1]
```diff
@@ -89,6 +89,9 @@ class BoltLink extends BoltActionElement {
       if (anchorTarget) {
         renderedLink.setAttribute('target', anchorTarget);
       }
+      if (renderedLink.getAttribute('target') === null && urlTarget) {
+        renderedLink.setAttribute('target', urlTarget);
+      }
```
Comment (remydenton) [inline packages/components/bolt-link/src/link.js]: This one isn't mentioned in the PR description and I don't really understand what it's doing.  I see `urlTarget` in `bolt-card-replacement-link`-- is this related to that?

Comment (sghoweri) [reply packages/components/bolt-link/src/link.js]: I'll get this removed -- that smells like leftover code from the Card work / merge and was already taken care of via the `if (anchorTarget)` logic right above it.

Comment (sghoweri) [reply packages/components/bolt-link/src/link.js]: All set!

File: packages/components/bolt-link/src/link.scss [hunk 2]
```diff
@@ -61,13 +61,18 @@ $bolt-link-spacing: 'xxsmall';
   }
 }
 
+.c-bolt-link--display-block {
+  display: block;
```
Comment (mikemai2awesome) [inline packages/components/bolt-link/src/link.scss]: Even though schema and CSS are added, there is no documentation. Let's add it to the display doc page.

Comment (sghoweri) [reply packages/components/bolt-link/src/link.scss]: Could you help with this?

Comment (mikemai2awesome) [reply packages/components/bolt-link/src/link.scss]: Doc page updated.

**Comment by sghoweri** [comment]
PR was released with v2.15.0
